src/rf/js/src/home/components/processing-block.js

```diff
@@ -4,38 +4,52 @@ var _ = require('underscore'),
     React = require('react');
 
 var LayerStatusComponent = React.createBackboneClass({
+    successClass: 'rf-icon-check',
+    pendingClass: 'rf-icon-loader',
+    workingClass: 'rf-icon-loader animate-spin',
+    failedClass: 'rf-icon-attention rf-failed text-danger',
+
     render: function() {
-        var checkClass = 'rf-icon-check',
-            spinnerClass = 'rf-icon-loader animate-spin',
-            failedClass = 'rf-icon-attention rf-failed text-danger',
-            preValidatedClass = spinnerClass,
-            processingClass = spinnerClass,
+        var preValidatedClass = this.workingClass,
+            validateClass = this.pendingClass,
+            thumbnailClass = this.pendingClass,
+            createWorkerClass = this.pendingClass,
+            chunkClass = this.pendingClass,
+            mosaicClass = this.pendingClass,
+            completeClass = this.pendingClass,
             actionLink = (<a href="#" className="text-danger">Cancel</a>),
             preValidatedErrorsExist = false,
             layerError = false,
             layerErrorComponent = (
                 <li>
-                    {this.getModel().get('error') ? this.getModel().get('error') : 'An unknown error occured.'}
+                    {this.getModel().get('failed_error') ? this.getModel().get('failed_error') : 'Processing failed.'}
                     <i className="rf-icon-attention"></i>
                 </li>
             ),
             preValidatedLabel = this.getModel().hasCopiedImages() ?
                 'Transferring Images' : 'Uploading Images';
 
-        if (this.getModel().isProcessing() ||
-            this.getModel().isCompleted()) {
-            preValidatedClass = checkClass;
+        if (this.getModel().get('status_upload_end') &&
+            this.getModel().get('status_upload_error') === null)
```

Comment:
Consider moving this into a method into the Layer model. Ditto for other stuff in this file that directly accesses the attributes in the Layer model.